src/geometry/GeometryEvaluator.cc

```diff
@@ -1330,6 +1334,60 @@ static std::unique_ptr<Geometry> rotatePolygon(const RotateExtrudeNode& node, co
   return builder.build();
 }
 
+static std::shared_ptr<const Geometry> rotatePolygon(const RotateExtrudeNode& node, const Polygon2d& poly){
```

Comment:
See #5096 - I suggest pulling out all the rotate_extrude tools into a new file, as this file is getting a bit hard to reason about.